shared/Microsoft.Extensions.WebEncoders.Sources/WebEncoders.cs

```diff
@@ -96,144 +184,193 @@ public static byte[] Base64UrlDecode(string input, int offset, int count)
         /// </remarks>
         public static byte[] Base64UrlDecode(string input, int offset, char[] buffer, int bufferOffset, int count)
         {
-            if (input == null)
-            {
-                throw new ArgumentNullException(nameof(input));
-            }
-            if (buffer == null)
-            {
-                throw new ArgumentNullException(nameof(buffer));
-            }
-
-            ValidateParameters(input.Length, nameof(input), offset, count);
-            if (bufferOffset < 0)
+            if (input == null
+                || (uint)offset > (uint)input.Length
+                || (uint)count > (uint)(input.Length - offset)
+                || buffer == null
+                || (uint)bufferOffset > (uint)buffer.Length)
             {
-                throw new ArgumentOutOfRangeException(nameof(bufferOffset));
+                ThrowInvalidArguments(input, offset, count, buffer, bufferOffset, validateBuffer: true);
             }
 
+            // Special-case empty input
             if (count == 0)
             {
                 return EmptyBytes;
             }
 
-            // Assumption: input is base64url encoded without padding and contains no whitespace.
-
-            var paddingCharsToAdd = GetNumBase64PaddingCharsToAddForDecode(count);
-            var arraySizeRequired = checked(count + paddingCharsToAdd);
-            Debug.Assert(arraySizeRequired % 4 == 0, "Invariant: Array length must be a multiple of 4.");
+            var base64Len = GetBufferSizeRequiredToUrlDecode(count, out int dataLength);
 
-            if (buffer.Length - bufferOffset < arraySizeRequired)
+            if ((uint)buffer.Length < (uint)(bufferOffset + base64Len))
```

Comment:
Any concerns here about integer overflow? `bufferOffset + base64Len` could be > int.MaxValue

Reply:
If so, the sum will be a negative value that is brought to very high values by the cast to uint, so the exception will be thrown. For me it seems OK. Am I wrong?

Change is motivated by replacing the subtraction with the addition.